src/Components/test/E2ETest/ServerRenderingTests/FormHandlingTests/FormWithParentBindingContextTest.cs

```diff
@@ -104,6 +104,24 @@ public void CanBindParameterToTheDefaultForm(bool suppressEnhancedNavigation)
         DispatchToFormCore(dispatchToForm);
     }
 
+    [Theory]
+    [InlineData(true)]
+    [InlineData(false)]
+    public void InputHiddenCanStoreData(bool suppressEnhancedNavigation)
+    {
+        var dispatchToForm = new DispatchToForm(this)
+        {
+            Url = "forms/default-form-input-hidden",
+            FormCssSelector = "form",
+            SuppressEnhancedNavigation = suppressEnhancedNavigation,
+        };
+        DispatchToFormCore(dispatchToForm);
+        var textHidden = Browser.Exists(By.Id("hidden")).GetDomProperty("value");
+        Assert.Equal("stranger", textHidden);
```

Comment:
This is fine but to make sure Selenium does not execute this check before the element is displayed, let's use a check that waits for the element with some default timeout, like here:

https://github.com/dariatiurina/aspnetcore/blob/e8fb9aa5e1437470c4a88c6ef3c8101c14df1529/src/Components/test/E2ETest/ServerRenderingTests/FormHandlingTests/FormWithParentBindingContextTest.cs#L1568